src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -351,20 +460,46 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
             // instead be the second mask. If so, skip the entire first mask and drain ASCII bytes
             // from the second mask.
 
-            if (currentMask == 0)
+            if (Sse2.IsSupported)
             {
-                pBuffer += SizeOfVector128;
-                currentMask = secondMask;
+                if (!ContainsNonAsciiByte_Sse2(currentSseMaskOrAdvSimdIndex))
```

Comment:
Is this check necessary? It looks like it's already being run before each `goto`.

Reply:
It is needed. The place that checks it is:
```c++
if (ContainsNonAsciiByte_Sse2(currentSseMaskOrAdvSimdIndex | secondSseMaskOrAdvSimdIndex))
{
  goto FoundNonAsciiDataInInnerLoop;
}
```

We need to use `secondSseMaskOrAdvSimdIndex` only if `currentSseMaskOrAdvSimdIndex` don't have non-ascii bytes.

Reply:
Yup, whatever Kunal said.